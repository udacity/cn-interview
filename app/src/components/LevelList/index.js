import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { LEVEL_ARRAY } from '../../utils/helper';

export default class LevelList extends Component {
  static propTypes = {
    onSelect: PropTypes.func,
    selected: PropTypes.string,
  };

  _Item = (level, selected, onSelect) => {
    const className = classNames('radio-item', {
      selected: level === selected,
    });
    return (
      <div className={className} onClick={() => onSelect(level)}>
        <h3 className="h-title h-radio">{level}</h3>
      </div>
    );
  };

  render() {
    const { onSelect, selected } = this.props;
    const levels = ['全部', ...LEVEL_ARRAY];
    return (
      <ul className="no-list-style">
        {levels.map(level => (
          <li key={level}>{this._Item(level, selected, onSelect)}</li>
        ))}
      </ul>
    );
  }
}
