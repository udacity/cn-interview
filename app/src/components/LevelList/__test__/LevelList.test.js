import React from 'react';
import { mount } from 'enzyme';
import LevelList from '..';

it('renders correctly', () => {
  const wrapper = mount(<LevelList selected="全部" />);
  expect(wrapper).toMatchSnapshot();
});

it('selected right item ', () => {
  const wrapper = mount(<LevelList selected="全部" />);
  expect(
    wrapper
      .find('.radio-item')
      .at(0)
      .hasClass('selected')
  ).toBe(true);
});

it('can trigger onSelect prop', () => {
  const handleSelect = jest.fn();
  const wrapper = mount(
    <LevelList selected="全部课程" onSelect={handleSelect} />
  );
  wrapper
    .find('.radio-item')
    .at(1)
    .simulate('click');
  expect(handleSelect).toHaveBeenCalled();
});
