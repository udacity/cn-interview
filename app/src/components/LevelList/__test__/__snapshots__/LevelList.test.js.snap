// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<LevelList
  selected="全部"
>
  <ul
    className="no-list-style"
  >
    <li
      key="全部"
    >
      <div
        className="radio-item selected"
        onClick={[Function]}
      >
        <h3
          className="h-title h-radio"
        >
          全部
        </h3>
      </div>
    </li>
    <li
      key="初级"
    >
      <div
        className="radio-item"
        onClick={[Function]}
      >
        <h3
          className="h-title h-radio"
        >
          初级
        </h3>
      </div>
    </li>
    <li
      key="中级"
    >
      <div
        className="radio-item"
        onClick={[Function]}
      >
        <h3
          className="h-title h-radio"
        >
          中级
        </h3>
      </div>
    </li>
    <li
      key="高级"
    >
      <div
        className="radio-item"
        onClick={[Function]}
      >
        <h3
          className="h-title h-radio"
        >
          高级
        </h3>
      </div>
    </li>
  </ul>
</LevelList>
`;
